casadi: 3.7.1 -> 3.7.2, drop abandoned clang_compiler interface, hpipm: unbreak casadi

nim65s opened the pull request:
- casadi: drop abandoned clang_compiler interface
ref. https://github.com/casadi/casadi/issues/4225#issuecomment-3352552113

- hpipm: unbreak casadi. 
I must have failed something while retesting after https://github.com/NixOS/nixpkgs/pull/444660#discussion_r2379636210

Now both seems OK on x86_64-linux & aarch64-darwin.

cc @vcunat  @LordGrimmauld 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nim65s committed:
casadi: drop abandoned clang_compiler interface

Co-authored-by: Luna <782440+LunNova@users.noreply.github.com>

nim65s committed:
hpipm: 0.1.3-unstable-2025-09-25 -> 0.1.3-unstable-2025-07-25

Back in time to unbreak casadi.
I must have failed something while retesting after https://github.com/NixOS/nixpkgs/pull/444660#discussion_r2379636210

nim65s committed:
casadi: add changelog

LunNova reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 447543`
Commit: `a0f63327f27c9a33aed537b934fc071a812166e9`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 57 packages built:</summary>
  <ul>
    <li>aligator</li>
    <li>aligator.doc</li>
    <li>casadi</li>
    <li>crocoddyl</li>
    <li>crocoddyl.doc</li>
    <li>haskellPackages.casadi-bindings-internal</li>
    <li>haskellPackages.casadi-bindings-internal.doc</li>
    <li>hpipm</li>
    <li>mim-solvers</li>
    <li>ndcurves</li>
    <li>ndcurves.doc</li>
    <li>pinocchio</li>
    <li>pinocchio.doc</li>
    <li>proxsuite-nlp</li>
    <li>proxsuite-nlp.doc</li>
    <li>python312Packages.aerosandbox</li>
    <li>python312Packages.aerosandbox.dist</li>
    <li>python312Packages.aligator</li>
    <li>python312Packages.aligator.doc</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.crocoddyl.doc</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.example-robot-data.doc</li>
    <li>python312Packages.mim-solvers</li>
    <li>python312Packages.ndcurves</li>
    <li>python312Packages.ndcurves.doc</li>
    <li>python312Packages.neuralfoil</li>
    <li>python312Packages.neuralfoil.dist</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.pinocchio.doc</li>
    <li>python312Packages.proxsuite-nlp</li>
    <li>python312Packages.proxsuite-nlp.doc</li>
    <li>python312Packages.tsid</li>
    <li>python312Packages.tsid.doc</li>
    <li>python313Packages.aerosandbox</li>
    <li>python313Packages.aerosandbox.dist</li>
    <li>python313Packages.aligator</li>
    <li>python313Packages.aligator.doc</li>
    <li>python313Packages.casadi</li>
    <li>python313Packages.crocoddyl</li>
    <li>python313Packages.crocoddyl.doc</li>
    <li>python313Packages.example-robot-data</li>
    <li>python313Packages.example-robot-data.doc</li>
    <li>python313Packages.mim-solvers</li>
    <li>python313Packages.ndcurves</li>
    <li>python313Packages.ndcurves.doc</li>
    <li>python313Packages.neuralfoil</li>
    <li>python313Packages.neuralfoil.dist</li>
    <li>python313Packages.pinocchio</li>
    <li>python313Packages.pinocchio.doc</li>
    <li>python313Packages.proxsuite-nlp</li>
    <li>python313Packages.proxsuite-nlp.doc</li>
    <li>python313Packages.tsid</li>
    <li>python313Packages.tsid.doc</li>
    <li>tsid</li>
    <li>tsid.doc</li>
  </ul>
</details>

nim65s committed:
casadi: 3.7.1 -> 3.7.2

Diff: https://github.com/casadi/casadi/compare/3.7.1...3.7.2

Changelog: https://github.com/casadi/casadi/releases/tag/3.7.2
